guidance_plugin_validator/src/guidance_plugin_validator/guidance_plugin_validator.py

```diff
@@ -0,0 +1,479 @@
+
+        # Read in config params
+        self.validation_duration = rospy.get_param('~validation_duration', 300) # Maximum time (sec) that node will spend conducting validation before results are considered final
+        self.startup_duration = rospy.get_param("~startup_duration", 90)
+        self.strategic_plugin_names = rospy.get_param('~guidance_strategic_plugins')
+        self.tactical_plugin_names = rospy.get_param('~guidance_tactical_plugins')
+        self.control_plugin_names = rospy.get_param('~guidance_control_plugins') 
+
+        # Write config params to log file
+        rospy.loginfo("Config params for guidance_plugin_validator:")
+        rospy.loginfo("Startup Duration: " + str(self.startup_duration) + " seconds")
+        rospy.loginfo("Validation Duration: " + str(self.validation_duration) + " seconds")
+        rospy.loginfo("Strategic Guidance Plugins: " + str(self.strategic_plugin_names))
+        rospy.loginfo("Tactical Guidance Plugins: " + str(self.tactical_plugin_names))
+        rospy.loginfo("Control Guidance Plugins: " + str(self.control_plugin_names))
+
+        # Boolean flag to indicate whether final results have been written to log file
+        self.has_logged_final_results = False
+
+        # Set spin rate
+        self.spin_rate = rospy.Rate(10) # 10 Hz
+        
+        # Initialize empty lists that will be populated with a <Plugin-Type>PluginResults object for each plugin
+        self.strategic_plugin_validation_results = [] # Populated with StrategicPluginResults objects
+        self.tactical_plugin_validation_results = []  # Populated with TacticalPluginResults objects
+        self.control_plugin_validation_results = []   # Populated with ControlPluginResults objects
+
+        # Initialize empty dicts that will be populated with each plugin's index in its respective 'validation results' list
+        self.strategic_plugin_results_index_by_name = {} # Key is plugin's name; Value is plugin's index in its respective 'validation results' list
+        self.tactical_plugin_results_index_by_name = {}  # Key is plugin's name; Value is plugin's index in its respective 'validation results' list
+        self.control_plugin_results_index_by_name = {}   # Key is plugin's name; Value is plugin's index in its respective 'validation results' list
+
+        # Call member function to populate the 'validation results' lists and 'index by name' dicts
+        self.populate_results_lists_and_dicts(self.strategic_plugin_names, self.tactical_plugin_names, self.control_plugin_names)
+
+    def populate_results_lists_and_dicts(self, strategic_plugin_names, tactical_plugin_names, control_plugin_names):
+        """Initialize the the 'validation results' lists and 'index by name' dicts for this object"""
+
+        # Populate list and dict for Strategic Plugins
+        for i in range(0, len(strategic_plugin_names)):
```

Comment:
I don't have a concrete comment here, but I feel there's got to be a more pythonic way to do this, not that I can think of it at the moment. Maybe ```for plugin in strategic_plugin_names:...``` and then index everything by name? Not sure, and doesn't have to be changed, but if wanted to put the thought out there into the ether.

Reply:
I have updated the implementation for this. The plugin now uses dicts with each key being the plugin's name, and the corresponding value being the plugin's <plugin-type>PluginResults object.